content/en/blog/_posts/2021-11-02-csi-migration-status.md

```diff
@@ -0,0 +1,122 @@
+
+- storage_operation_duration_seconds
+  - This metric is exposed from the Kube-Controller-Manager. A new field `migrated` is added to the metric to indicate whether this storage operation is a CSI migration operation.
+- csi_sidecar_operations_seconds
+  - This metric is exposed from the CSI sidecars(csi-provisioner, csi-attacher, csi-resizer). A new field `migrated` is added to the metric to indicate whether this storage operation is a CSI migration operation.
+- csi_kubelet_operations_seconds
+  - This metric is exposed from Kubelet. It exposes CSI operation, driver  name, status and `migrated` field to indicate whether this storage operation is a CSI migration operation.
+
+### Bug fixes and feature improvement
+
+We have been able to identify bugs like dangling attachment, garbage collecting, incorrect topology label ... and we have fixed all of them to make the feature more health.
+
+### Cloud Provider && Cluster Lifecycle Collaboration
+
+We, Sig-Storage, has been working closely with Sig-CloudProvider and Sig-ClusterLifeCycle to provide a better experience for the feature launching.
+
+If you are a user from public cloud provider managed Kubernetes service like GKE, EKS, AKS, there is nothing need to be done from your end. The cloud providers will manage the feature gate and ensure the feature lands correctly.
+
+If you are a user of some other kubernetes distribution services, check the official documentations to get support for the feature. We have communicate with the Sig-ClusterLifeCycle to request early planing for CSI Migration GA and they have been working hard to ensure a smooth experience.
+
+## What is the timeline/status?
+
+The current and targeted releases for each individual cloud driver is shown in the table below:
+
+| Driver            | Alpha         | Beta (in-tree deprecated) | Beta (on-by-default) | GA            | Target "in-tree plugin" removal |
+| ----------------- | ------------- | ------------------------- | -------------------- | ------------- | ------------------------------- |
+| AWS EBS           | 1.14          | 1.17                      | 1.23                 | 1.24 (Target) | 1.26 (Target)                   |
+| GCE PD            | 1.14          | 1.17                      | 1.23                 | 1.24 (Target) | 1.26 (Target)                   |
+| OpenStack Cinder  | 1.14          | 1.18                      | 1.21                 | 1.24 (Target) | 1.26 (Target)                   |
+| Azure Disk        | 1.15          | 1.19                      | 1.23                 | 1.24 (Target) | 1.26 (Target)                   |
+| Azure File        | 1.15          | 1.21                      | 1.23                 | 1.24 (Target) | 1.26 (Target)                   |
+| VSphere           | 1.18          | 1.19                      | 1.23 (Potentially miss)| 1.24 (Target) | 1.26 (Target)                   |
```

Comment:
The blog will be published after 1.23 is released so "(Potentially miss)" can be removed.

Reply:
I will remove it for now and wait for the final status of vSphere to see if this need to be changed.